test/integration/local/test_python_service.py

```diff
@@ -0,0 +1,124 @@
+# language governing permissions and limitations under the License.
+
+import os
+import shutil
+import subprocess
+import sys
+import time
+
+import pytest
+
+import requests
+
+PING_URL = 'http://localhost:8080/ping'
+INVOCATIONS_URL = 'http://localhost:8080/invocations'
+
+
+@pytest.fixture(scope='session', autouse=True, params=['input_output_handler', 'handler'])
+def volume(tmpdir_factory, request):
+    try:
+        script_file = 'test/integration/local/script/{}_example.py'.format(request.param)
+        requirements_file = 'test/integration/local/script/requirements_example.txt'
+        shutil.copy(script_file, tmpdir_factory.getbasetemp().join('inference.py'))
+        shutil.copy(requirements_file, tmpdir_factory.getbasetemp().join('requirements.txt'))
+
+        model_dir = os.path.abspath('test/resources/models')
+        subprocess.check_call(
+            'docker volume create --name model_inference_volume --opt type=none '
+            '--opt device={} --opt o=bind'.format(model_dir).split())
+        yield model_dir
+    finally:
+        subprocess.check_call('docker volume prune -f'.split())
+
+
+@pytest.fixture(scope='module', autouse=True, params=['1.11', '1.12'])
+def container(request):
+    try:
+        command = (
+            'docker run --name sagemaker-tensorflow-serving-test -p 8080:8080'
+            ' --mount type=volume,source=model_inference_volume,target=/opt/ml/model'
+            ' -e SAGEMAKER_TFS_DEFAULT_MODEL_NAME=half_plus_three'
+            ' -e SAGEMAKER_TFS_NGINX_LOGLEVEL=info'
+            ' -e SAGEMAKER_BIND_TO_PORT=8080'
+            ' -e SAGEMAKER_SAFE_PORT_RANGE=9000-9999'
+            ' sagemaker-tensorflow-serving:{}-cpu serve'
+        ).format(request.param)
+
+        proc = subprocess.Popen(command.split(), stdout=sys.stdout, stderr=subprocess.STDOUT)
+
+        attempts = 0
+        while attempts < 5:
+            time.sleep(3)
+            try:
+                requests.get('http://localhost:8080/ping')
+                break
+            except:
+                attempts += 1
+                pass
+
+        yield proc.pid
+    finally:
+        subprocess.check_call('docker rm -f sagemaker-tensorflow-serving-test'.split())
+
+
+def make_headers(content_type, method):
+    headers = {
+        'Content-Type': content_type,
+        'X-Amzn-SageMaker-Custom-Attributes': 'tfs-model-name=half_plus_three,tfs-method=%s' % method
+    }
+    return headers
+
+
+def test_predict_json():
+    headers = make_headers('application/json', 'predict')
+    data = '{"instances": [1.0, 2.0, 5.0]}'
+    response = requests.post(INVOCATIONS_URL, data=data, headers=headers).json()
+    assert response == {'predictions': [3.5, 4.0, 5.5]}
+
+
+def test_zero_content():
+    headers = make_headers('application/json', 'predict')
+    data = ''
+    response = requests.post(INVOCATIONS_URL, data=data, headers=headers)
+    assert response.status_code != 200
```

Comment:
let's assert the expected failure code (and message if there is one) instead of allowing any kind of failure. as it is, the test will hide an unexpected failure.